airbyte-cdk/bulk/core/load/src/main/kotlin/io/airbyte/cdk/load/state/PipelineEventBookkeepingRouter.kt

```diff
@@ -177,6 +177,11 @@ class PipelineEventBookkeepingRouter(
                         val totalCounts = streamWithKeyAndCount.sumOf { (_, count) -> count }
                         Pair(singleKey, totalCounts)
                     } else {
+                        val sourceCounts = checkpoint.sourceStats?.recordCount ?: 0L
```

Comment:
checking: we don't want to throw here? I.e. if somehow a sync is running in speed mode, but the source doesn't attach a record count - we'd prefer to continue the sync, rather than treat it as an error?